### PR TITLE
simplify with dict comprehensions

### PR DESCRIPTION
**Pull request by cclauss**

### CODE REVIEW

File: release/__init__.py [hunk 1]
```diff
@@ -79,12 +75,8 @@ def strip_locals(data):
...
 
     if isinstance(data, dict):
-        data = copy.copy(data)
-        for k in set(data.keys()):
-            if isinstance(k, str) and k.startswith('local_'):
-                del data[k]
-            else:
-                data[k] = strip_locals(data[k])
+        return {key: strip_locals(value) for key, value in data.items()
+                if not (isinstance(k, str) and k.startswith('local_'))}
```
Comment (cmaloney) [inline release/__init__.py]: k needs to be changed to key. If you run `tox -e py34-unittests` to run the tests appliacble to this file, a python NameError exception will get thrown here 

(I'm currently working on docs for how to run code quality checks among other things)

**Comment by cmaloney** [comment]
You should be able to click the button to see the CI / Pull Request status failures and login as guest now

**Comment by cmaloney** [comment]
Looks good to me / :ship: Do you want to rebase on master and squash into a single commit, or would you rather I do it?

**Comment by cclauss** [comment]
Please do it...  I need to sleep... Thx

**Comment by cmaloney** [comment]
Squashed and merged: https://github.com/dcos/dcos/commit/02e8c6ec57c98e2751e4e42f6007263cf0ea91ac

**Pull request closed without merging**